File: src/main/java/com/example/demo/model/Athlete.java
package com.example.demo.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "athletes")
@Getter
@Setter
public class Athlete {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false, unique = true)
    private String email;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", unique = true)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "coach_id")
    private Coach coach;
}

File: crossfit_flutter/lib/models/athlete.dart
class Athlete {
  final int id;
  final String name;
  final String email;
  final int? coachId;
  final String? coachName;

  Athlete({
    required this.id,
    required this.name,
    required this.email,
    this.coachId,
    this.coachName,
  });

  factory Athlete.fromJson(Map<String, dynamic> json) {
    return Athlete(
      id: json['id'] as int,
      name: json['name'] as String,
      email: json['email'] as String,
      coachId: json['coachId'] as int?,
      coachName: json['coachName'] as String?,
    );
  }
}
